posthog/session_recordings/session_recording_api.py

```diff
@@ -1001,32 +998,57 @@ def _stream_blob_v2_to_client(
+            async def fetch_single_block_async(block_index: int) -> tuple[int, str | None]:
+                try:
                     block = blocks[block_index]
-                    try:
-                        decompressed_blocks.append(session_recording_v2_object_storage.client().fetch_block(block.url))
-                    except BlockFetchError:
-                        logger.exception(
-                            "Failed to fetch block",
-                            recording_id=recording.session_id,
-                            team_id=self.team.id,
-                            block_index=block_index,
-                        )
-                        raise exceptions.APIException("Failed to load recording block")
+                    content = await asyncio.to_thread(
+                        session_recording_v2_object_storage.client().fetch_block, block.url
+                    )
+                    return block_index, content
+                except BlockFetchError:
+                    logger.exception(
+                        "Failed to fetch block",
+                        recording_id=recording.session_id,
+                        team_id=self.team.id,
+                        block_index=block_index,
+                    )
+                    return block_index, None
+
+            with timer("fetch_blocks_parallel__stream_blob_v2_to_client"):
+                tasks = [fetch_single_block_async(block_index) for block_index in range(min_blob_key, max_blob_key + 1)]
+
+                results = await asyncio.gather(*tasks)
+
+                decompressed_blocks: list[str | None] = [None] * len(results)
+                block_errors: list[int] = []
+
+                for block_index, content in results:
+                    if content is None:
+                        block_errors.append(block_index)
+                    else:
+                        decompressed_blocks[block_index - min_blob_key] = content
 
+            if block_errors:
+                raise exceptions.APIException("Failed to load recording block")
+
+            # After error check, all values are guaranteed to be strings
             response = HttpResponse(
-                content="\n".join(decompressed_blocks),
+                content="\n".join(cast(list[str], decompressed_blocks)),
                 content_type="application/jsonl",
             )
-
-            # Set caching headers - blocks are immutable, so we can cache for a while
             response["Cache-Control"] = "max-age=3600"
             response["Content-Disposition"] = "inline"
-
             return response
 
+    def _stream_blob_v2_to_client(
+        self,
+        recording: SessionRecording,
+        timer: ServerTimingsGathered,
+        min_blob_key: int,
+        max_blob_key: int,
+    ) -> HttpResponse:
+        return asyncio.run(self._stream_blob_v2_to_client_async(recording, timer, min_blob_key, max_blob_key))
```

Comment:
Django, technically, should support async views (instead of `asyncio.run`), but I wasn't able to make it work with PostHog myself, so I think `run` should be ok.